src/test/javascript/spec/component/overview/course-conversations/course-conversations.component.spec.ts

```diff
@@ -155,6 +147,57 @@ examples.forEach((activeConversation) => {
             expect(component.activeConversation).toEqual(activeConversation);
         });
 
+        describe('Dialog Opening', () => {
+            const expectedResults = [undefined, true];
+            const mockModalRef: Partial<NgbModalRef> = {
+                componentInstance: {
+                    course: {},
+                    initialize: jest.fn(),
+                },
+            };
+
+            beforeEach(() => {
+                // Set a mock course with id 1
+                component.course = { id: 1 } as Course;
+            });
+
+            it('should open the group chat creation dialog', fakeAsync(() => {
+                mockModalRef.result = Promise.resolve(undefined);
+                const spy = jest.spyOn(modalService, 'open').mockReturnValue(mockModalRef as NgbModalRef);
+                component.onCreateGroupChatPressed();
+                tick();
+                expect(spy).toHaveBeenCalledWith(GroupChatCreateDialogComponent, expect.anything());
```

Comment:
Could we be more specific here?
```suggestion
                expect(spy).toHaveBeenCalledExactlyOnceWith(GroupChatCreateDialogComponent, expect.anything());
```